### DISCUSSION TITLE
Lerna version is generating a weird tag and I cannot understand why

### DISCUSSION THREAD

**Discussion by marcocastignoli** [lerna/lerna#3820] [Q&A] [answered]
* I'm testing on this project: https://github.com/marcocastignoli/sourcify
* my lerna config has the `"version": "independent"` setting.
* I ran this at the beginning `npx lerna exec --concurrency 1 --stream -- 'conventional-changelog --preset angular --release-count 0 --commit-path $PWD --pkg $PWD/package.json --outfile $PWD/CHANGELOG.md --verbose --lerna-package $LERNA_PACKAGE_NAME'`
* when I run `npx lerna version --no-push` for the first time, I select for each package the patch option
```
lerna notice cli v7.1.5
lerna info versioning independent
lerna info Assuming all packages changed
lerna info version rooted leaf detected, skipping synthetic root lifecycles
? Select a new version for sourcify-server (currently 1.0.0) Patch (1.0.1)
? Select a new version for @ethereum-sourcify/bytecode-utils (currently 1.1.1) Patch (1.1.2)
? Select a new version for @ethereum-sourcify/contract-call-decoder (currently 0.2.2) Patch (0.2.3)
? Select a new version for @ethereum-sourcify/lib-sourcify (currently 1.3.1) Patch (1.3.2)
? Select a new version for sourcify-ui (currently 0.1.0) Patch (0.1.1)

Changes:
 - sourcify-server: 1.0.0 => 1.0.1
 - @ethereum-sourcify/bytecode-utils: 1.1.1 => 1.1.2
 - @ethereum-sourcify/contract-call-decoder: 0.2.2 => 0.2.3
 - @ethereum-sourcify/lib-sourcify: 1.3.1 => 1.3.2
 - sourcify-ui: 0.1.0 => 0.1.1 (private)
 ```
* the repo becomes like this:
<img width="631" alt="Screenshot 2023-08-29 alle 10 26 09" src="https://github.com/lerna/lerna/assets/10226180/baeb819c-f1ef-4309-8995-84d5a7ef9e7b">

questions:
* I cannot understand where that 1.4.0 version is coming from.
* Why two separate commits?
* then it pushes in the packages/lib-sourcify's CHANGELOG the changelog that should be in the root CHANGELOG
* I tried many things but without success

Thanks in advance for the answers :)

**Comment by marcocastignoli** [comment]
Additional important note.

in the following interaction I selected the patch version 1.4.1 but it automatically bumped the version first from 1.4.0 to 1.4.1 and then to 1.5.0!

```
marcocastignoli@Air-di-Marco sourcify % npx lerna version --amend --conventional-commits=false  --no-changelog
lerna notice cli v7.1.5
lerna info versioning independent
lerna info Looking for changed packages since sourcify-server@1.0.2
lerna info ignoring diff in paths matching [ '*.md' ]
lerna info version rooted leaf detected, skipping synthetic root lifecycles
lerna WARN version Skipping working tree validation, proceed at your own risk
? Select a new version for sourcify-server (currently 1.0.2) Patch (1.0.3)
? Select a new version for @ethereum-sourcify/lib-sourcify (currently 1.4.0) Patch (1.4.1)

Changes:
 - sourcify-server: 1.0.2 => 1.0.3
 - @ethereum-sourcify/lib-sourcify: 1.4.0 => 1.4.1

? Are you sure you want to create these versions? Yes
lerna info execute Skipping git push
lerna info execute Skipping releases
lerna info lifecycle @ethereum-sourcify/lib-sourcify@1.4.1~version: @ethereum-sourcify/lib-sourcify@1.4.1

> @ethereum-sourcify/lib-sourcify@1.4.1 version /Users/marcocastignoli/Projects/marcocastignoli/sourcify/packages/lib-sourcify
> standard-version

✔ bumping version in package.json from 1.4.1 to 1.5.0
✔ bumping version in package-lock.json from 1.4.1 to 1.5.0
✔ outputting changes to CHANGELOG.md
✔ committing package-lock.json and package.json and CHANGELOG.md
✔ tagging release v1.5.0
ℹ Run `git push --follow-tags origin staging && npm publish` to publish
```

**Comment by fahslaj** [comment, marked as answer]
@marcocastignoli Your `lib-sourcify` package has a script in `package.json` named "version" that is being executed since "version" is a [Lerna lifecycle script](https://github.com/lerna/lerna/tree/main/libs/commands/version#lifecycle-scripts). This additional bump is happening via the `standard-version` tool that is being executed within that script. You can see its output in the logging you provided above as well:

```
> @ethereum-sourcify/lib-sourcify@1.4.1 version /Users/marcocastignoli/Projects/marcocastignoli/sourcify/packages/lib-sourcify
> standard-version

✔ bumping version in package.json from 1.4.1 to 1.5.0
✔ bumping version in package-lock.json from 1.4.1 to 1.5.0
✔ outputting changes to CHANGELOG.md
✔ committing package-lock.json and package.json and CHANGELOG.md
✔ tagging release v1.5.0
ℹ Run `git push --follow-tags origin staging && npm publish` to publish
```

To prevent this, just remove the `version` script from that package or rename it to something that isn't a lifecycle script name.

**Reply by marcocastignoli** [reply]
Thanks a lot for the answer, you saved me from hours of debugging 🙏
